tldextract/tldextract.py

```diff
@@ -187,8 +187,15 @@ def __call__(self, url):
             .rstrip(".")
 
         is_punycode = netloc.startswith('xn--') or '.xn--' in netloc
+        # Some IDN urls might generate an error
         if is_punycode:
-            netloc = codecs.decode(netloc.encode('ascii'), 'idna')
+            try:
+                netloc = codecs.decode(netloc.encode('ascii'), 'idna')
+            except UnicodeError:
+                LOG.warn('UnicodeError when decoding netloc ' + netloc)
```

Comment:
By the way, that rant above had nothing to do with the PR really. I was just thinking out loud. Documented that thinking in the FAQ in 12f43eb3b3c94f3dcf071b4d826fdc609f4f00cf.

Again, thanks for the fix!